src/test/java/uk/gov/hmcts/reform/professionalapi/controller/internal/OrganisationInternalControllerTest.java

```diff
@@ -300,4 +306,29 @@ public void testInviteUserToOrganisation() throws JsonProcessingException {
         verify(organisationServiceMock, times(1)).getOrganisationByOrgIdentifier(orgId);
         verify(professionalUserServiceMock, times(1)).findProfessionalUserByEmailAddress("some@email.com");
     }
+
+    @Test
+    public void testDeleteOrganisation() {
+
+        final HttpStatus expectedHttpStatus = HttpStatus.NO_CONTENT;
+        String orgId = UUID.randomUUID().toString().substring(0, 7);
+        organisation.setStatus(OrganisationStatus.PENDING);
+
+        when(organisationServiceMock.getOrganisationByOrgIdentifier(orgId)).thenReturn(organisation);
+        when(organisationServiceMock.deleteOrganisation(organisation)).thenReturn(deleteOrganisationResponse);
+        ResponseEntity<?> actual = organisationInternalController.deleteOrganisation(orgId);
+
+        verify(organisationServiceMock, times(1)).getOrganisationByOrgIdentifier(orgId);
+        verify(organisationServiceMock, times(1)).deleteOrganisation(organisation);
+
+        assertThat(actual).isNotNull();
+        assertThat(actual.getStatusCode()).isEqualTo(expectedHttpStatus);
+    }
+
+    @Test(expected = EmptyResultDataAccessException.class)
+    public void testDeleteOrganisationThrows404WhenNoOrgFound() {
+        String orgId = UUID.randomUUID().toString().substring(0, 7);
+        when(organisationServiceMock.getOrganisationByOrgIdentifier(orgId)).thenReturn(null);
```

Comment:
is it possible to add a `verify()` for this `when()`?

Reply:
added the entry for verify